pkg/lint/linter/linter.go

```diff
@@ -11,3 +13,23 @@ type Linter interface {
 	Name() string
 	Desc() string
 }
+
+type Noop struct {
+	name string
+	desc string
+	run  func(pass *analysis.Pass) (interface{}, error)
+}
+
+func (n Noop) Run(_ context.Context, lintCtx *Context) ([]result.Issue, error) {
+	lintCtx.Log.Warnf("%s is disabled because of go1.18."+
+		" If you are not using go1.18, you can set `go: go1.17` in the `run` section.", n.name)
+	return nil, nil
+}
```

Comment:
Do you think it would add any value to link the tracking issue (#2649) in the output? I'm thinking it may avoid new issues being created by people who don't bother searching existing ones.